ui/app/AppLayouts/Chat/popups/community/BackUpCommunityPopup.qml

```diff
@@ -0,0 +1,150 @@
+import QtQuick 2.12
+import QtQuick.Controls 2.14
+import QtQuick.Layouts 1.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Core.Utils 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Popups 0.1
+
+import utils 1.0
+import shared.controls 1.0
+import shared 1.0
+
+StatusStackModal {
+    id: root
+
+    property var store
+    property string privateKey
+    property int maximumLength: 4
+
+    width: 480
+    height: 504
+    header.title: (currentIndex === 0) ? qsTr("Back up community key")
+                                       : qsTr("Confirm your community key")
+    padding: Style.current.padding
+
+    stackItems: [
+    ColumnLayout {
+        spacing: Style.current.halfPadding
+
+        StatusInput {
+            id: pKeyInput
+            Layout.fillWidth: true
+            leftPadding: 14
+            rightPadding: Style.current.halfPadding
+            topPadding: 0
+            bottomPadding: 0
+            label: qsTr("Your community key")
+            minimumHeight: 56
+            maximumHeight: 56
+            input.text: root.privateKey
+            input.edit.readOnly: true
+            input.rightComponent: StatusButton {
+                anchors.verticalCenter: parent.verticalCenter
+                borderColor: Theme.palette.primaryColor1
+                size: StatusBaseButton.Size.Tiny
+                text: qsTr("Copy")
+                onClicked: {
+                    text = qsTr("Copied");
+                    root.store.copyToClipboard(root.privateKey);
+                }
+            }
+        }
+
+        StatusBaseText {
+            Layout.fillWidth: true
+            Layout.preferredHeight: 36
+            Layout.topMargin: Style.current.halfPadding
+            text: qsTr("You should keep it safe and only share it with people you trust to take ownership of your community")
+            wrapMode: Text.WordWrap
+            font.pixelSize: 13
+            color: Theme.palette.baseColor1
+        }
+
+        StatusBaseText {
+            Layout.fillWidth: true
+            Layout.preferredHeight: 34
+            text: qsTr("You can also use this key to import your community on another device")
+            wrapMode: Text.WordWrap
+            font.pixelSize: 13
+            color: Theme.palette.baseColor1
+        }
+        },
+        Item {
+            TextEdit {
+                id: validationInput
+                property string previousText: text
+                width: parent.width
+                height: 24
+                anchors.centerIn: parent
+                anchors.verticalCenterOffset: -indicationText.height
+                horizontalAlignment: Text.AlignHCenter
+                color: Theme.palette.baseColor1
+                cursorDelegate: Rectangle {
```

Comment:
BTW this is like the 3rd place we have this "cursor delegate"; might be a good idea to extract it to a separate component

Reply:
done